Add Unique Region Names SE

pragasette opened the pull request:
* Add location
    https://www.nexusmods.com/skyrimspecialedition/mods/56452/

* Add message
    Patch provided for Realistic Water Two

* Add message
    Patch provided for Realistic Water Two and Cutting Room Floor

* Add message
    Patch provided for Realistic Water Two and MEZF

* Add requirement
    Required by Regional Save Names

MacSplody commented on masterlist.yaml:
```diff
@@ -13180,7 +13187,27 @@ plugins:
+        condition: 'active("RealisticWaterTwo - Unique Region Names - MissingEZs_AllExteriors.esp")'
+  - name: 'Unique Region Names.esp'
+    url: [ 'https://www.nexusmods.com/skyrimspecialedition/mods/56452/' ]
+    group: *cellSettingsGroup
```
It shouldn't be put this late, it wipes outs changes from too many other mods.
The previous version I made myself was stripped down to avoid that exact problem.

```suggestion
```

pragasette replied:
Yes, I wasn't very conservative here: some tests I took in my load orders made it look good enough, but I trust your judgement.

MacSplody commented on masterlist.yaml:
```diff
@@ -1839,6 +1839,13 @@ globals:
       - '[.NET Script Framework](https://www.nexusmods.com/skyrimspecialedition/mods/21294/)'
     condition: 'file("DLLPlugins/NetScriptFramework.Runtime.dll") and ((not file("../binkw64_.dll") and not checksum("../binkw64.dll", 6C2F54AB)) and (not file("../d3dx9_42.dll")))'
 
+# Other Missing Requirements
+  - <<: *missingRequirementXforY
+    subs:
+      - '[Unique Region Names SE](https://www.nexusmods.com/skyrimspecialedition/mods/56452/)'
+      - '[Regional Save Names](https://www.nexusmods.com/skyrimspecialedition/mods/49698/)'
+    condition: 'file("SKSE/Plugins/po3_RegionalSaveNames.dll") and not active("Unique Region Names.esp")'
+
```
Globals are usually for messages where there no plugin provided with the mod.

```suggestion
```

---

If you don't want to use the `req:` key, then you can use a `*requiresX` message attached to `Unique Region Names.esp`.

```yaml
  - name: 'Unique Region Names.esp'
    msg:
      - <<: *requiresX
        subs: [ '[Regional Save Names](https://www.nexusmods.com/skyrimspecialedition/mods/49698/)' ]
        condition: 'file("SKSE/Plugins/po3_RegionalSaveNames.dll")'
```

pragasette replied:
This works perfectly, I had wrongly assumed this would only be evaluated when the listed plugin was present and `globals` was the use case for this. If I used the `req` key, wouldn't it make Unique Region Names depend on Regional Save Names?

MacSplody commented on masterlist.yaml:
```diff
@@ -13180,7 +13187,27 @@ plugins:
+        condition: 'active("RealisticWaterTwo.esp") and active("Cutting Room Floor.esp") and not active("RealisticWaterTwo - Unique Region Names - Cutting Room Floor.esp")'
+      - <<: *patchProvided
+        subs: [ 'Realistic Water Two SE and Missing Encounter Zones FIXED' ]
+        condition: 'active("RealisticWaterTwo.esp") and active("MissingEZs_AllExteriors.esp") and not active("RealisticWaterTwo - Unique Region Names - MissingEZs_AllExteriors.esp")'
```
Add newlines between files from different mods as well.

```suggestion
        condition: 'active("RealisticWaterTwo.esp") and active("MissingEZs_AllExteriors.esp") and not active("RealisticWaterTwo - Unique Region Names - MissingEZs_AllExteriors.esp")'

```

MacSplody commented on masterlist.yaml:
```diff
@@ -13180,7 +13187,27 @@ plugins:
   - name: 'Better Descriptions SSE - YOT Patch.esp'
     req: [ 'YOT - Your Own Thoughts.esp' ]
-
+  - name: 'RealisticWaterTwo - Unique Region Names.esp'
```
```suggestion

  - name: 'RealisticWaterTwo - Unique Region Names.esp'
```

pragasette replied:
Now I see the logic, I had assumed the existing blank line was a typo.

MacSplody reviewed (changes requested):
Some changes required.

pragasette commented:
Thank you for the thorough review, I amended the changes as per your recommendations and I'm adding some comments above after yours.

MacSplody commented on masterlist.yaml:
```diff
@@ -13181,6 +13181,30 @@ plugins:
   - name: 'Better Descriptions SSE - YOT Patch.esp'
     req: [ 'YOT - Your Own Thoughts.esp' ]
 
+  - name: 'RealisticWaterTwo - Unique Region Names.esp'
+    msg:
+      - <<: *alreadyInX
+        subs: [ 'RealisticWaterTwo - Unique Region Names - Cutting Room Floor.esp' ]
+        condition: 'active("RealisticWaterTwo - Unique Region Names - Cutting Room Floor.esp")'
+      - <<: *alreadyInX
+        subs: [ 'RealisticWaterTwo - Unique Region Names - MissingEZs_AllExteriors.esp' ]
+        condition: 'active("RealisticWaterTwo - Unique Region Names - MissingEZs_AllExteriors.esp")'
+  - name: 'Unique Region Names.esp'
+    url: [ 'https://www.nexusmods.com/skyrimspecialedition/mods/56452/' ]
+    msg:
+      - <<: *patchProvided
+        subs: [ 'Realistic Water Two SE' ]
+        condition: 'active("RealisticWaterTwo.esp") and not active("Cutting Room Floor.esp") and not active("MissingEZs_AllExteriors.esp") and not active("RealisticWaterTwo - Unique Region Names.esp") and not active("RealisticWaterTwo - Unique Region Names - (Cutting Room Floor|MissingEZs_AllExteriors)\.esp")'
+      - <<: *patchProvided
+        subs: [ 'Realistic Water Two SE and Cutting Room Floor SSE' ]
+        condition: 'active("RealisticWaterTwo.esp") and active("Cutting Room Floor.esp") and not active("RealisticWaterTwo - Unique Region Names - Cutting Room Floor.esp")'
+      - <<: *patchProvided
+        subs: [ 'Realistic Water Two SE and Missing Encounter Zones FIXED' ]
+        condition: 'active("RealisticWaterTwo.esp") and active("MissingEZs_AllExteriors.esp") and not active("RealisticWaterTwo - Unique Region Names - MissingEZs_AllExteriors.esp")'
+      - <<: *requiresX
+        subs: [ '[Regional Save Names](https://www.nexusmods.com/skyrimspecialedition/mods/49698/)' ]
+        condition: 'file("SKSE/Plugins/po3_RegionalSaveNames.dll")'
```
Just one more change, keep patches/add-ons hosted on the same page directly after the main file(s).

```suggestion
  - name: 'Unique Region Names.esp'
    url: [ 'https://www.nexusmods.com/skyrimspecialedition/mods/56452/' ]
    msg:
      - <<: *patchProvided
        subs: [ 'Realistic Water Two SE' ]
        condition: 'active("RealisticWaterTwo.esp") and not active("Cutting Room Floor.esp") and not active("MissingEZs_AllExteriors.esp") and not active("RealisticWaterTwo - Unique Region Names.esp") and not active("RealisticWaterTwo - Unique Region Names - (Cutting Room Floor|MissingEZs_AllExteriors)\.esp")'
      - <<: *patchProvided
        subs: [ 'Realistic Water Two SE and Cutting Room Floor SSE' ]
        condition: 'active("RealisticWaterTwo.esp") and active("Cutting Room Floor.esp") and not active("RealisticWaterTwo - Unique Region Names - Cutting Room Floor.esp")'
      - <<: *patchProvided
        subs: [ 'Realistic Water Two SE and Missing Encounter Zones FIXED' ]
        condition: 'active("RealisticWaterTwo.esp") and active("MissingEZs_AllExteriors.esp") and not active("RealisticWaterTwo - Unique Region Names - MissingEZs_AllExteriors.esp")'
      - <<: *requiresX
        subs: [ '[Regional Save Names](https://www.nexusmods.com/skyrimspecialedition/mods/49698/)' ]
        condition: 'file("SKSE/Plugins/po3_RegionalSaveNames.dll")'
  - name: 'RealisticWaterTwo - Unique Region Names.esp'
    msg:
      - <<: *alreadyInX
        subs: [ 'RealisticWaterTwo - Unique Region Names - Cutting Room Floor.esp' ]
        condition: 'active("RealisticWaterTwo - Unique Region Names - Cutting Room Floor.esp")'
      - <<: *alreadyInX
        subs: [ 'RealisticWaterTwo - Unique Region Names - MissingEZs_AllExteriors.esp' ]
        condition: 'active("RealisticWaterTwo - Unique Region Names - MissingEZs_AllExteriors.esp")'
```

pragasette replied:
Sure, it makes sense (I had sorted them alphabetically), amended.

MacSplody reviewed (changes requested):
Just one more format change required.

pragasette committed:
Add Unique Region Names SE

* Add location
 - https://www.nexusmods.com/skyrimspecialedition/mods/56452/

* Add message
 - Patch provided for Realistic Water Two

* Add message
 - Patch provided for Realistic Water Two and Cutting Room Floor

* Add message
 - Patch provided for Realistic Water Two and MEZF

* Add requirement
 - Required by Regional Save Names

MacSplody commented:
@pragasette Thanks.
